src/org/centos/contra/Infra/Utils.groovy

```diff
@@ -304,65 +311,71 @@ def createOpenstackInstances(HashMap<String, String>openstackData){
  * @param command - The linchpin command to execute. Typically this should be "up" or "destroy"
  * @param options - Options to be passed to the linchpin command
  * @param verbose - whether to include -vvv to the execution of the linchpin command
- * @param linchpinContainerName - name of linchpin container in executing pod. default is 'linchpin-executor'.
+ * @param containerName - name of linchpin container in executing pod. default is 'linchpin-executor'.
+ * @param args    - Arguments for the linchpin command. For linchpin "up" or "destroy" this could be target.
  * @return
  */
-def executeInLinchpin(String command, String options, Boolean verbose, String linchpinContainerName){
-
-    String containerName = linchpinContainerName ?: 'linchpin-executor'
+def executeInLinchpin(String command, String options, Boolean verbose, String containerName, String args = '') {
+    containerName = containerName ?: 'linchpin-executor'
```

Comment:
Should be either:
`def containerName = containerName ?: 'linchpin-executor'`
or (preferably)
`String containerName = containerName ?: 'linchpin-executor'`

Reply:
Wouldn't that be confusing since containerName is already defined in the function as one of the parameters? Or should I define a new variable so that I can explicitly typecast it?

Reply:
It's less of an explicity typecast thing, and more of a Groovy semantics thing.
http://groovy-lang.org/semantics.html#_variable_definition

Reply:
Ah I see - the goal here is to treat the param as read only. I will clean it up accordingly. :+1:

Reply:
Taking a step back, we could actually leave it as it is.

Reply:
I went ahead and assigned it a new a variable. One more address on the stack shouldn't get us into too much trouble. :) It also looks cuter next to the declaration below it. :smile_cat: